{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 0: Import and Download NCBI Taxa Database\n",
    "\n",
    "reference:http://etetoolkit.org/docs/latest/tutorial/tutorial_ncbitaxonomy.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from ete3 import NCBITaxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download NCBI Taxa Database\n",
    "ncbi = NCBITaxa()\n",
    "#ncbi.update_taxonomy_database()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 1: Read Nanopore Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['BC01', 'BC10', 'BC12', 'BC02', 'BC11', 'BC08', 'BC06', 'BC04',\n",
       "       'BC09', 'BC05', 'BC07', 'BC03', nan], dtype=object)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_path = '../data/191115/12_gut_test_classification.csv'\n",
    "result_path = '../result/191115/'\n",
    "\n",
    "raw = pd.read_csv('../data/191115/12_gut_test_classification.csv', encoding = 'utf-8')\n",
    "\n",
    "############################\n",
    "# how many diferrent barcode\n",
    "\n",
    "raw['barcode'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>read_id</th>\n",
       "      <th>exit_status</th>\n",
       "      <th>runid</th>\n",
       "      <th>taxid</th>\n",
       "      <th>barcode</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>lca</th>\n",
       "      <th>genus_taxid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastq_runid_0d12eab41107590678909ab4b985f43e38...</td>\n",
       "      <td>00cd5abd-7e2e-4327-8c30-ffbf7f38e0bf</td>\n",
       "      <td>Classification successful</td>\n",
       "      <td>0d12eab41107590678909ab4b985f43e389bdb15</td>\n",
       "      <td>186802</td>\n",
       "      <td>BC01</td>\n",
       "      <td>77.55</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>fastq_runid_0d12eab41107590678909ab4b985f43e38...</td>\n",
       "      <td>011a6dd8-ac78-4aea-8876-77a11002d63d</td>\n",
       "      <td>Classification successful</td>\n",
       "      <td>0d12eab41107590678909ab4b985f43e389bdb15</td>\n",
       "      <td>541000</td>\n",
       "      <td>BC10</td>\n",
       "      <td>87.34</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>fastq_runid_0d12eab41107590678909ab4b985f43e38...</td>\n",
       "      <td>0268997c-6c0f-4c29-85e0-7876956c1518</td>\n",
       "      <td>Classification successful</td>\n",
       "      <td>0d12eab41107590678909ab4b985f43e389bdb15</td>\n",
       "      <td>186803</td>\n",
       "      <td>BC10</td>\n",
       "      <td>94.40</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fastq_runid_0d12eab41107590678909ab4b985f43e38...</td>\n",
       "      <td>02d1af6d-3a03-4b76-a3af-d9106058d522</td>\n",
       "      <td>Classification successful</td>\n",
       "      <td>0d12eab41107590678909ab4b985f43e389bdb15</td>\n",
       "      <td>186802</td>\n",
       "      <td>BC01</td>\n",
       "      <td>86.73</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fastq_runid_0d12eab41107590678909ab4b985f43e38...</td>\n",
       "      <td>030f2253-419c-4bb7-8d5d-998ddabbe851</td>\n",
       "      <td>Classification successful</td>\n",
       "      <td>0d12eab41107590678909ab4b985f43e389bdb15</td>\n",
       "      <td>1411144</td>\n",
       "      <td>BC12</td>\n",
       "      <td>85.47</td>\n",
       "      <td>0</td>\n",
       "      <td>816</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            filename  \\\n",
       "0  fastq_runid_0d12eab41107590678909ab4b985f43e38...   \n",
       "1  fastq_runid_0d12eab41107590678909ab4b985f43e38...   \n",
       "2  fastq_runid_0d12eab41107590678909ab4b985f43e38...   \n",
       "3  fastq_runid_0d12eab41107590678909ab4b985f43e38...   \n",
       "4  fastq_runid_0d12eab41107590678909ab4b985f43e38...   \n",
       "\n",
       "                                read_id                exit_status  \\\n",
       "0  00cd5abd-7e2e-4327-8c30-ffbf7f38e0bf  Classification successful   \n",
       "1  011a6dd8-ac78-4aea-8876-77a11002d63d  Classification successful   \n",
       "2  0268997c-6c0f-4c29-85e0-7876956c1518  Classification successful   \n",
       "3  02d1af6d-3a03-4b76-a3af-d9106058d522  Classification successful   \n",
       "4  030f2253-419c-4bb7-8d5d-998ddabbe851  Classification successful   \n",
       "\n",
       "                                      runid    taxid barcode  accuracy  lca  \\\n",
       "0  0d12eab41107590678909ab4b985f43e389bdb15   186802    BC01     77.55    1   \n",
       "1  0d12eab41107590678909ab4b985f43e389bdb15   541000    BC10     87.34    1   \n",
       "2  0d12eab41107590678909ab4b985f43e389bdb15   186803    BC10     94.40    1   \n",
       "3  0d12eab41107590678909ab4b985f43e389bdb15   186802    BC01     86.73    1   \n",
       "4  0d12eab41107590678909ab4b985f43e389bdb15  1411144    BC12     85.47    0   \n",
       "\n",
       "   genus_taxid  \n",
       "0           -1  \n",
       "1           -1  \n",
       "2           -1  \n",
       "3           -1  \n",
       "4          816  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BC01 read count: 90502\n",
      "BC02 read count: 96384\n",
      "BC03 read count: 13559\n",
      "BC04 read count: 65033\n",
      "BC05 read count: 114032\n",
      "BC06 read count: 111664\n",
      "BC07 read count: 27294\n",
      "BC08 read count: 79342\n",
      "BC09 read count: 81997\n",
      "BC10 read count: 80869\n",
      "BC11 read count: 114584\n",
      "BC12 read count: 147803\n"
     ]
    }
   ],
   "source": [
    "#################################################\n",
    "# Select raw data \n",
    "# (1) 'exit_status' == 'Classification successful'\n",
    "# (2) 'barcode' == BC01, BC02, BC03, BC04\n",
    "\n",
    "raw = pd.read_csv(file_path, encoding = 'utf-8')\n",
    "\n",
    "\n",
    "#####\n",
    "#BC01\n",
    "#####\n",
    "raw_sucess_BC01 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC01')]\n",
    "\n",
    "raw_sucess_BC01.to_csv(result_path +'raw_sucess_BC01.csv', encoding='utf-8')\n",
    "print('BC01 read count:', len(raw_sucess_BC01))\n",
    "\n",
    "\n",
    "#####\n",
    "#BC02\n",
    "#####\n",
    "raw_sucess_BC02 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC02')]\n",
    "\n",
    "raw_sucess_BC02.to_csv(result_path +'raw_sucess_BC02.csv', encoding='utf-8')\n",
    "print('BC02 read count:',len(raw_sucess_BC02))\n",
    "\n",
    "\n",
    "#####\n",
    "#BC03\n",
    "#####\n",
    "raw_sucess_BC03 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC03')]\n",
    "\n",
    "raw_sucess_BC03.to_csv(result_path +'raw_sucess_BC03.csv', encoding='utf-8')\n",
    "print('BC03 read count:',len(raw_sucess_BC03))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC04\n",
    "#####\n",
    "raw_sucess_BC04 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC04')]\n",
    "\n",
    "raw_sucess_BC04.to_csv(result_path +'raw_sucess_BC04.csv', encoding='utf-8')\n",
    "print('BC04 read count:',len(raw_sucess_BC04))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC05\n",
    "#####\n",
    "raw_sucess_BC05 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC05')]\n",
    "\n",
    "raw_sucess_BC05.to_csv(result_path +'raw_sucess_BC05.csv', encoding='utf-8')\n",
    "print('BC05 read count:',len(raw_sucess_BC05))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC06\n",
    "#####\n",
    "raw_sucess_BC06 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC06')]\n",
    "\n",
    "raw_sucess_BC06.to_csv(result_path +'raw_sucess_BC06.csv', encoding='utf-8')\n",
    "print('BC06 read count:',len(raw_sucess_BC06))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC07\n",
    "#####\n",
    "raw_sucess_BC07 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC07')]\n",
    "\n",
    "raw_sucess_BC07.to_csv(result_path +'raw_sucess_BC07.csv', encoding='utf-8')\n",
    "print('BC07 read count:',len(raw_sucess_BC07))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC08\n",
    "#####\n",
    "raw_sucess_BC08 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC08')]\n",
    "\n",
    "raw_sucess_BC08.to_csv(result_path +'raw_sucess_BC08.csv', encoding='utf-8')\n",
    "print('BC08 read count:',len(raw_sucess_BC08))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC09\n",
    "#####\n",
    "raw_sucess_BC09 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC09')]\n",
    "\n",
    "raw_sucess_BC09.to_csv(result_path +'raw_sucess_BC09.csv', encoding='utf-8')\n",
    "print('BC09 read count:',len(raw_sucess_BC09))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC10\n",
    "#####\n",
    "raw_sucess_BC10 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC10')]\n",
    "\n",
    "raw_sucess_BC10.to_csv(result_path +'raw_sucess_BC10.csv', encoding='utf-8')\n",
    "print('BC10 read count:',len(raw_sucess_BC10))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC11\n",
    "#####\n",
    "raw_sucess_BC11 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC11')]\n",
    "\n",
    "raw_sucess_BC11.to_csv(result_path +'raw_sucess_BC11.csv', encoding='utf-8')\n",
    "print('BC11 read count:',len(raw_sucess_BC11))\n",
    "\n",
    "\n",
    "\n",
    "#####\n",
    "#BC12\n",
    "#####\n",
    "raw_sucess_BC12 = raw[(raw['exit_status'] == 'Classification successful') & (raw['barcode'] == 'BC12')]\n",
    "\n",
    "raw_sucess_BC12.to_csv(result_path +'raw_sucess_BC12.csv', encoding='utf-8')\n",
    "print('BC12 read count:',len(raw_sucess_BC12))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finish\n"
     ]
    }
   ],
   "source": [
    "# BC01 accurate rate\n",
    "\n",
    "BC01_total = len(raw_sucess_BC01)\n",
    "\n",
    "BC01_99 = len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 99)])\n",
    "BC01_95 = len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 95)])\n",
    "BC01_90 = len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 90)])\n",
    "BC01_85 = len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 85)])\n",
    "BC01_80 = len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 80)])\n",
    "BC01_75 = len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 75)])\n",
    "\n",
    "BC01_99_per = (len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 99)]) / len(raw_sucess_BC01)) * 100\n",
    "BC01_95_per = (len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 95)]) / len(raw_sucess_BC01)) * 100\n",
    "BC01_90_per = (len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 90)]) / len(raw_sucess_BC01)) * 100\n",
    "BC01_85_per = (len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 85)]) / len(raw_sucess_BC01)) * 100\n",
    "BC01_80_per = (len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 80)]) / len(raw_sucess_BC01)) * 100\n",
    "BC01_75_per = (len(raw_sucess_BC01[(raw_sucess_BC01['accuracy'] > 75)]) / len(raw_sucess_BC01)) * 100\n",
    "\n",
    "\n",
    "# BC02 accurate rate\n",
    "\n",
    "BC02_total = len(raw_sucess_BC02)\n",
    "\n",
    "BC02_99 = len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 99)])\n",
    "BC02_95 = len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 95)])\n",
    "BC02_90 = len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 90)])\n",
    "BC02_85 = len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 85)])\n",
    "BC02_80 = len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 80)])\n",
    "BC02_75 = len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 75)])\n",
    "\n",
    "BC02_99_per = (len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 99)]) / len(raw_sucess_BC02)) * 100\n",
    "BC02_95_per = (len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 95)]) / len(raw_sucess_BC02)) * 100\n",
    "BC02_90_per = (len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 90)]) / len(raw_sucess_BC02)) * 100\n",
    "BC02_85_per = (len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 85)]) / len(raw_sucess_BC02)) * 100\n",
    "BC02_80_per = (len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 80)]) / len(raw_sucess_BC02)) * 100\n",
    "BC02_75_per = (len(raw_sucess_BC02[(raw_sucess_BC02['accuracy'] > 75)]) / len(raw_sucess_BC02)) * 100\n",
    "\n",
    "\n",
    "# BC03 accurate rate\n",
    "\n",
    "BC03_total = len(raw_sucess_BC03)\n",
    "\n",
    "BC03_99 = len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 99)])\n",
    "BC03_95 = len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 95)])\n",
    "BC03_90 = len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 90)])\n",
    "BC03_85 = len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 85)])\n",
    "BC03_80 = len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 80)])\n",
    "BC03_75 = len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 75)])\n",
    "\n",
    "BC03_99_per = (len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 99)]) / len(raw_sucess_BC03)) * 100\n",
    "BC03_95_per = (len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 95)]) / len(raw_sucess_BC03)) * 100\n",
    "BC03_90_per = (len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 90)]) / len(raw_sucess_BC03)) * 100\n",
    "BC03_85_per = (len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 85)]) / len(raw_sucess_BC03)) * 100\n",
    "BC03_80_per = (len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 80)]) / len(raw_sucess_BC03)) * 100\n",
    "BC03_75_per = (len(raw_sucess_BC03[(raw_sucess_BC03['accuracy'] > 75)]) / len(raw_sucess_BC03)) * 100\n",
    "\n",
    "\n",
    "# BC04 accurate rate\n",
    "\n",
    "BC04_total = len(raw_sucess_BC04)\n",
    "\n",
    "BC04_99 = len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 99)])\n",
    "BC04_95 = len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 95)])\n",
    "BC04_90 = len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 90)])\n",
    "BC04_85 = len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 85)])\n",
    "BC04_80 = len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 80)])\n",
    "BC04_75 = len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 75)])\n",
    "\n",
    "BC04_99_per = (len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 99)]) / len(raw_sucess_BC04)) * 100\n",
    "BC04_95_per = (len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 95)]) / len(raw_sucess_BC04)) * 100\n",
    "BC04_90_per = (len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 90)]) / len(raw_sucess_BC04)) * 100\n",
    "BC04_85_per = (len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 85)]) / len(raw_sucess_BC04)) * 100\n",
    "BC04_80_per = (len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 80)]) / len(raw_sucess_BC04)) * 100\n",
    "BC04_75_per = (len(raw_sucess_BC04[(raw_sucess_BC04['accuracy'] > 75)]) / len(raw_sucess_BC04)) * 100\n",
    "\n",
    "\n",
    "# BC05 accurate rate\n",
    "\n",
    "BC05_total = len(raw_sucess_BC05)\n",
    "\n",
    "BC05_99 = len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 99)])\n",
    "BC05_95 = len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 95)])\n",
    "BC05_90 = len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 90)])\n",
    "BC05_85 = len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 85)])\n",
    "BC05_80 = len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 80)])\n",
    "BC05_75 = len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 75)])\n",
    "\n",
    "BC05_99_per = (len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 99)]) / len(raw_sucess_BC05)) * 100\n",
    "BC05_95_per = (len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 95)]) / len(raw_sucess_BC05)) * 100\n",
    "BC05_90_per = (len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 90)]) / len(raw_sucess_BC05)) * 100\n",
    "BC05_85_per = (len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 85)]) / len(raw_sucess_BC05)) * 100\n",
    "BC05_80_per = (len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 80)]) / len(raw_sucess_BC05)) * 100\n",
    "BC05_75_per = (len(raw_sucess_BC05[(raw_sucess_BC05['accuracy'] > 75)]) / len(raw_sucess_BC05)) * 100\n",
    "\n",
    "\n",
    "\n",
    "# BC06 accurate rate\n",
    "\n",
    "BC06_total = len(raw_sucess_BC06)\n",
    "\n",
    "BC06_99 = len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 99)])\n",
    "BC06_95 = len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 95)])\n",
    "BC06_90 = len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 90)])\n",
    "BC06_85 = len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 85)])\n",
    "BC06_80 = len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 80)])\n",
    "BC06_75 = len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 75)])\n",
    "\n",
    "BC06_99_per = (len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 99)]) / len(raw_sucess_BC06)) * 100\n",
    "BC06_95_per = (len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 95)]) / len(raw_sucess_BC06)) * 100\n",
    "BC06_90_per = (len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 90)]) / len(raw_sucess_BC06)) * 100\n",
    "BC06_85_per = (len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 85)]) / len(raw_sucess_BC06)) * 100\n",
    "BC06_80_per = (len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 80)]) / len(raw_sucess_BC06)) * 100\n",
    "BC06_75_per = (len(raw_sucess_BC06[(raw_sucess_BC06['accuracy'] > 75)]) / len(raw_sucess_BC06)) * 100\n",
    "\n",
    "\n",
    "# BC07 accurate rate\n",
    "\n",
    "BC07_total = len(raw_sucess_BC07)\n",
    "\n",
    "BC07_99 = len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 99)])\n",
    "BC07_95 = len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 95)])\n",
    "BC07_90 = len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 90)])\n",
    "BC07_85 = len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 85)])\n",
    "BC07_80 = len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 80)])\n",
    "BC07_75 = len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 75)])\n",
    "\n",
    "BC07_99_per = (len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 99)]) / len(raw_sucess_BC07)) * 100\n",
    "BC07_95_per = (len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 95)]) / len(raw_sucess_BC07)) * 100\n",
    "BC07_90_per = (len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 90)]) / len(raw_sucess_BC07)) * 100\n",
    "BC07_85_per = (len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 85)]) / len(raw_sucess_BC07)) * 100\n",
    "BC07_80_per = (len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 80)]) / len(raw_sucess_BC07)) * 100\n",
    "BC07_75_per = (len(raw_sucess_BC07[(raw_sucess_BC07['accuracy'] > 75)]) / len(raw_sucess_BC07)) * 100\n",
    "\n",
    "\n",
    "# BC08 accurate rate\n",
    "\n",
    "BC08_total = len(raw_sucess_BC08)\n",
    "\n",
    "BC08_99 = len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 99)])\n",
    "BC08_95 = len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 95)])\n",
    "BC08_90 = len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 90)])\n",
    "BC08_85 = len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 85)])\n",
    "BC08_80 = len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 80)])\n",
    "BC08_75 = len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 75)])\n",
    "\n",
    "BC08_99_per = (len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 99)]) / len(raw_sucess_BC08)) * 100\n",
    "BC08_95_per = (len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 95)]) / len(raw_sucess_BC08)) * 100\n",
    "BC08_90_per = (len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 90)]) / len(raw_sucess_BC08)) * 100\n",
    "BC08_85_per = (len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 85)]) / len(raw_sucess_BC08)) * 100\n",
    "BC08_80_per = (len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 80)]) / len(raw_sucess_BC08)) * 100\n",
    "BC08_75_per = (len(raw_sucess_BC08[(raw_sucess_BC08['accuracy'] > 75)]) / len(raw_sucess_BC08)) * 100\n",
    "\n",
    "\n",
    "# BC09 accurate rate\n",
    "\n",
    "BC09_total = len(raw_sucess_BC09)\n",
    "\n",
    "BC09_99 = len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 99)])\n",
    "BC09_95 = len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 95)])\n",
    "BC09_90 = len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 90)])\n",
    "BC09_85 = len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 85)])\n",
    "BC09_80 = len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 80)])\n",
    "BC09_75 = len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 75)])\n",
    "\n",
    "BC09_99_per = (len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 99)]) / len(raw_sucess_BC09)) * 100\n",
    "BC09_95_per = (len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 95)]) / len(raw_sucess_BC09)) * 100\n",
    "BC09_90_per = (len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 90)]) / len(raw_sucess_BC09)) * 100\n",
    "BC09_85_per = (len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 85)]) / len(raw_sucess_BC09)) * 100\n",
    "BC09_80_per = (len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 80)]) / len(raw_sucess_BC09)) * 100\n",
    "BC09_75_per = (len(raw_sucess_BC09[(raw_sucess_BC09['accuracy'] > 75)]) / len(raw_sucess_BC09)) * 100\n",
    "\n",
    "\n",
    "# BC10 accurate rate\n",
    "\n",
    "BC10_total = len(raw_sucess_BC10)\n",
    "\n",
    "BC10_99 = len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 99)])\n",
    "BC10_95 = len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 95)])\n",
    "BC10_90 = len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 90)])\n",
    "BC10_85 = len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 85)])\n",
    "BC10_80 = len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 80)])\n",
    "BC10_75 = len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 75)])\n",
    "\n",
    "BC10_99_per = (len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 99)]) / len(raw_sucess_BC10)) * 100\n",
    "BC10_95_per = (len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 95)]) / len(raw_sucess_BC10)) * 100\n",
    "BC10_90_per = (len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 90)]) / len(raw_sucess_BC10)) * 100\n",
    "BC10_85_per = (len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 85)]) / len(raw_sucess_BC10)) * 100\n",
    "BC10_80_per = (len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 80)]) / len(raw_sucess_BC10)) * 100\n",
    "BC10_75_per = (len(raw_sucess_BC10[(raw_sucess_BC10['accuracy'] > 75)]) / len(raw_sucess_BC10)) * 100\n",
    "\n",
    "\n",
    "# BC11 accurate rate\n",
    "\n",
    "BC11_total = len(raw_sucess_BC11)\n",
    "\n",
    "BC11_99 = len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 99)])\n",
    "BC11_95 = len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 95)])\n",
    "BC11_90 = len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 90)])\n",
    "BC11_85 = len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 85)])\n",
    "BC11_80 = len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 80)])\n",
    "BC11_75 = len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 75)])\n",
    "\n",
    "BC11_99_per = (len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 99)]) / len(raw_sucess_BC11)) * 100\n",
    "BC11_95_per = (len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 95)]) / len(raw_sucess_BC11)) * 100\n",
    "BC11_90_per = (len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 90)]) / len(raw_sucess_BC11)) * 100\n",
    "BC11_85_per = (len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 85)]) / len(raw_sucess_BC11)) * 100\n",
    "BC11_80_per = (len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 80)]) / len(raw_sucess_BC11)) * 100\n",
    "BC11_75_per = (len(raw_sucess_BC11[(raw_sucess_BC11['accuracy'] > 75)]) / len(raw_sucess_BC11)) * 100\n",
    "\n",
    "\n",
    "# BC12 accurate rate\n",
    "\n",
    "BC12_total = len(raw_sucess_BC12)\n",
    "\n",
    "BC12_99 = len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 99)])\n",
    "BC12_95 = len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 95)])\n",
    "BC12_90 = len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 90)])\n",
    "BC12_85 = len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 85)])\n",
    "BC12_80 = len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 80)])\n",
    "BC12_75 = len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 75)])\n",
    "\n",
    "BC12_99_per = (len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 99)]) / len(raw_sucess_BC12)) * 100\n",
    "BC12_95_per = (len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 95)]) / len(raw_sucess_BC12)) * 100\n",
    "BC12_90_per = (len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 90)]) / len(raw_sucess_BC12)) * 100\n",
    "BC12_85_per = (len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 85)]) / len(raw_sucess_BC12)) * 100\n",
    "BC12_80_per = (len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 80)]) / len(raw_sucess_BC12)) * 100\n",
    "BC12_75_per = (len(raw_sucess_BC12[(raw_sucess_BC12['accuracy'] > 75)]) / len(raw_sucess_BC12)) * 100\n",
    "\n",
    "print('finish')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BC01</th>\n",
       "      <th>BC02</th>\n",
       "      <th>BC03</th>\n",
       "      <th>BC04</th>\n",
       "      <th>BC05</th>\n",
       "      <th>BC06</th>\n",
       "      <th>BC07</th>\n",
       "      <th>BC08</th>\n",
       "      <th>BC09</th>\n",
       "      <th>BC10</th>\n",
       "      <th>BC11</th>\n",
       "      <th>BC12</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Barcode total count</th>\n",
       "      <td>90502.000000</td>\n",
       "      <td>96384.000000</td>\n",
       "      <td>13559.000000</td>\n",
       "      <td>65033.000000</td>\n",
       "      <td>114032.000000</td>\n",
       "      <td>111664.000000</td>\n",
       "      <td>27294.000000</td>\n",
       "      <td>79342.000000</td>\n",
       "      <td>81997.000000</td>\n",
       "      <td>80869.000000</td>\n",
       "      <td>114584.000000</td>\n",
       "      <td>147803.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 99%</th>\n",
       "      <td>24.000000</td>\n",
       "      <td>45.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>11.000000</td>\n",
       "      <td>43.000000</td>\n",
       "      <td>38.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>23.000000</td>\n",
       "      <td>20.000000</td>\n",
       "      <td>14.000000</td>\n",
       "      <td>34.000000</td>\n",
       "      <td>29.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 95%</th>\n",
       "      <td>1718.000000</td>\n",
       "      <td>4440.000000</td>\n",
       "      <td>928.000000</td>\n",
       "      <td>5883.000000</td>\n",
       "      <td>8753.000000</td>\n",
       "      <td>10176.000000</td>\n",
       "      <td>3741.000000</td>\n",
       "      <td>3952.000000</td>\n",
       "      <td>8188.000000</td>\n",
       "      <td>3082.000000</td>\n",
       "      <td>5621.000000</td>\n",
       "      <td>5051.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 90%</th>\n",
       "      <td>16864.000000</td>\n",
       "      <td>41468.000000</td>\n",
       "      <td>8877.000000</td>\n",
       "      <td>45244.000000</td>\n",
       "      <td>77960.000000</td>\n",
       "      <td>84891.000000</td>\n",
       "      <td>23296.000000</td>\n",
       "      <td>44279.000000</td>\n",
       "      <td>63750.000000</td>\n",
       "      <td>41275.000000</td>\n",
       "      <td>77759.000000</td>\n",
       "      <td>65003.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 85%</th>\n",
       "      <td>47749.000000</td>\n",
       "      <td>77543.000000</td>\n",
       "      <td>12889.000000</td>\n",
       "      <td>63269.000000</td>\n",
       "      <td>108956.000000</td>\n",
       "      <td>109151.000000</td>\n",
       "      <td>27082.000000</td>\n",
       "      <td>71831.000000</td>\n",
       "      <td>80615.000000</td>\n",
       "      <td>72510.000000</td>\n",
       "      <td>109331.000000</td>\n",
       "      <td>121973.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 80%</th>\n",
       "      <td>85000.000000</td>\n",
       "      <td>94571.000000</td>\n",
       "      <td>13539.000000</td>\n",
       "      <td>65006.000000</td>\n",
       "      <td>113919.000000</td>\n",
       "      <td>111614.000000</td>\n",
       "      <td>27288.000000</td>\n",
       "      <td>79022.000000</td>\n",
       "      <td>81968.000000</td>\n",
       "      <td>80494.000000</td>\n",
       "      <td>114442.000000</td>\n",
       "      <td>144912.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above75%</th>\n",
       "      <td>90502.000000</td>\n",
       "      <td>96384.000000</td>\n",
       "      <td>13559.000000</td>\n",
       "      <td>65033.000000</td>\n",
       "      <td>114032.000000</td>\n",
       "      <td>111664.000000</td>\n",
       "      <td>27294.000000</td>\n",
       "      <td>79342.000000</td>\n",
       "      <td>81997.000000</td>\n",
       "      <td>80869.000000</td>\n",
       "      <td>114584.000000</td>\n",
       "      <td>147803.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 99% percentage</th>\n",
       "      <td>0.026519</td>\n",
       "      <td>0.046688</td>\n",
       "      <td>0.014750</td>\n",
       "      <td>0.016914</td>\n",
       "      <td>0.037709</td>\n",
       "      <td>0.034031</td>\n",
       "      <td>0.010991</td>\n",
       "      <td>0.028988</td>\n",
       "      <td>0.024391</td>\n",
       "      <td>0.017312</td>\n",
       "      <td>0.029673</td>\n",
       "      <td>0.019621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 95% percentage</th>\n",
       "      <td>1.898301</td>\n",
       "      <td>4.606574</td>\n",
       "      <td>6.844163</td>\n",
       "      <td>9.046177</td>\n",
       "      <td>7.675916</td>\n",
       "      <td>9.113053</td>\n",
       "      <td>13.706309</td>\n",
       "      <td>4.980968</td>\n",
       "      <td>9.985731</td>\n",
       "      <td>3.811102</td>\n",
       "      <td>4.905571</td>\n",
       "      <td>3.417387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 90% percentage</th>\n",
       "      <td>18.633842</td>\n",
       "      <td>43.023738</td>\n",
       "      <td>65.469430</td>\n",
       "      <td>69.570833</td>\n",
       "      <td>68.366774</td>\n",
       "      <td>76.023607</td>\n",
       "      <td>85.352092</td>\n",
       "      <td>55.807769</td>\n",
       "      <td>77.746747</td>\n",
       "      <td>51.039335</td>\n",
       "      <td>67.862005</td>\n",
       "      <td>43.979486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 85% percentage</th>\n",
       "      <td>52.760160</td>\n",
       "      <td>80.452150</td>\n",
       "      <td>95.058633</td>\n",
       "      <td>97.287531</td>\n",
       "      <td>95.548618</td>\n",
       "      <td>97.749498</td>\n",
       "      <td>99.223273</td>\n",
       "      <td>90.533387</td>\n",
       "      <td>98.314572</td>\n",
       "      <td>89.663530</td>\n",
       "      <td>95.415590</td>\n",
       "      <td>82.524035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above 80% percentage</th>\n",
       "      <td>93.920576</td>\n",
       "      <td>98.118982</td>\n",
       "      <td>99.852496</td>\n",
       "      <td>99.958483</td>\n",
       "      <td>99.900905</td>\n",
       "      <td>99.955223</td>\n",
       "      <td>99.978017</td>\n",
       "      <td>99.596683</td>\n",
       "      <td>99.964633</td>\n",
       "      <td>99.536287</td>\n",
       "      <td>99.876073</td>\n",
       "      <td>98.044018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>above75% percentage</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              BC01          BC02          BC03          BC04  \\\n",
       "Barcode total count   90502.000000  96384.000000  13559.000000  65033.000000   \n",
       "above 99%                24.000000     45.000000      2.000000     11.000000   \n",
       "above 95%              1718.000000   4440.000000    928.000000   5883.000000   \n",
       "above 90%             16864.000000  41468.000000   8877.000000  45244.000000   \n",
       "above 85%             47749.000000  77543.000000  12889.000000  63269.000000   \n",
       "above 80%             85000.000000  94571.000000  13539.000000  65006.000000   \n",
       "above75%              90502.000000  96384.000000  13559.000000  65033.000000   \n",
       "above 99% percentage      0.026519      0.046688      0.014750      0.016914   \n",
       "above 95% percentage      1.898301      4.606574      6.844163      9.046177   \n",
       "above 90% percentage     18.633842     43.023738     65.469430     69.570833   \n",
       "above 85% percentage     52.760160     80.452150     95.058633     97.287531   \n",
       "above 80% percentage     93.920576     98.118982     99.852496     99.958483   \n",
       "above75% percentage     100.000000    100.000000    100.000000    100.000000   \n",
       "\n",
       "                               BC05           BC06          BC07  \\\n",
       "Barcode total count   114032.000000  111664.000000  27294.000000   \n",
       "above 99%                 43.000000      38.000000      3.000000   \n",
       "above 95%               8753.000000   10176.000000   3741.000000   \n",
       "above 90%              77960.000000   84891.000000  23296.000000   \n",
       "above 85%             108956.000000  109151.000000  27082.000000   \n",
       "above 80%             113919.000000  111614.000000  27288.000000   \n",
       "above75%              114032.000000  111664.000000  27294.000000   \n",
       "above 99% percentage       0.037709       0.034031      0.010991   \n",
       "above 95% percentage       7.675916       9.113053     13.706309   \n",
       "above 90% percentage      68.366774      76.023607     85.352092   \n",
       "above 85% percentage      95.548618      97.749498     99.223273   \n",
       "above 80% percentage      99.900905      99.955223     99.978017   \n",
       "above75% percentage      100.000000     100.000000    100.000000   \n",
       "\n",
       "                              BC08          BC09          BC10           BC11  \\\n",
       "Barcode total count   79342.000000  81997.000000  80869.000000  114584.000000   \n",
       "above 99%                23.000000     20.000000     14.000000      34.000000   \n",
       "above 95%              3952.000000   8188.000000   3082.000000    5621.000000   \n",
       "above 90%             44279.000000  63750.000000  41275.000000   77759.000000   \n",
       "above 85%             71831.000000  80615.000000  72510.000000  109331.000000   \n",
       "above 80%             79022.000000  81968.000000  80494.000000  114442.000000   \n",
       "above75%              79342.000000  81997.000000  80869.000000  114584.000000   \n",
       "above 99% percentage      0.028988      0.024391      0.017312       0.029673   \n",
       "above 95% percentage      4.980968      9.985731      3.811102       4.905571   \n",
       "above 90% percentage     55.807769     77.746747     51.039335      67.862005   \n",
       "above 85% percentage     90.533387     98.314572     89.663530      95.415590   \n",
       "above 80% percentage     99.596683     99.964633     99.536287      99.876073   \n",
       "above75% percentage     100.000000    100.000000    100.000000     100.000000   \n",
       "\n",
       "                               BC12  \n",
       "Barcode total count   147803.000000  \n",
       "above 99%                 29.000000  \n",
       "above 95%               5051.000000  \n",
       "above 90%              65003.000000  \n",
       "above 85%             121973.000000  \n",
       "above 80%             144912.000000  \n",
       "above75%              147803.000000  \n",
       "above 99% percentage       0.019621  \n",
       "above 95% percentage       3.417387  \n",
       "above 90% percentage      43.979486  \n",
       "above 85% percentage      82.524035  \n",
       "above 80% percentage      98.044018  \n",
       "above75% percentage      100.000000  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = {'BC01':[BC01_total, BC01_99, BC01_95, BC01_90, BC01_85, BC01_80, BC01_75, BC01_99_per, BC01_95_per, BC01_90_per, BC01_85_per, BC01_80_per, BC01_75_per],\n",
    "        'BC02':[BC02_total, BC02_99, BC02_95, BC02_90, BC02_85, BC02_80, BC02_75, BC02_99_per, BC02_95_per, BC02_90_per, BC02_85_per, BC02_80_per, BC02_75_per],\n",
    "        'BC03':[BC03_total, BC03_99, BC03_95, BC03_90, BC03_85, BC03_80, BC03_75, BC03_99_per, BC03_95_per, BC03_90_per, BC03_85_per, BC03_80_per, BC03_75_per],\n",
    "        'BC04':[BC04_total, BC04_99, BC04_95, BC04_90, BC04_85, BC04_80, BC04_75, BC04_99_per, BC04_95_per, BC04_90_per, BC04_85_per, BC04_80_per, BC04_75_per],\n",
    "        'BC05':[BC05_total, BC05_99, BC05_95, BC05_90, BC05_85, BC05_80, BC05_75, BC05_99_per, BC05_95_per, BC05_90_per, BC05_85_per, BC05_80_per, BC05_75_per],\n",
    "        'BC06':[BC06_total, BC06_99, BC06_95, BC06_90, BC06_85, BC06_80, BC06_75, BC06_99_per, BC06_95_per, BC06_90_per, BC06_85_per, BC06_80_per, BC06_75_per],\n",
    "        'BC07':[BC07_total, BC07_99, BC07_95, BC07_90, BC07_85, BC07_80, BC07_75, BC07_99_per, BC07_95_per, BC07_90_per, BC07_85_per, BC07_80_per, BC07_75_per],\n",
    "        'BC08':[BC08_total, BC08_99, BC08_95, BC08_90, BC08_85, BC08_80, BC08_75, BC08_99_per, BC08_95_per, BC08_90_per, BC08_85_per, BC08_80_per, BC08_75_per],\n",
    "        'BC09':[BC09_total, BC09_99, BC09_95, BC09_90, BC09_85, BC09_80, BC09_75, BC09_99_per, BC09_95_per, BC09_90_per, BC09_85_per, BC09_80_per, BC09_75_per],\n",
    "        'BC10':[BC10_total, BC10_99, BC10_95, BC10_90, BC10_85, BC10_80, BC10_75, BC10_99_per, BC10_95_per, BC10_90_per, BC10_85_per, BC10_80_per, BC10_75_per],\n",
    "        'BC11':[BC11_total, BC11_99, BC11_95, BC11_90, BC11_85, BC11_80, BC11_75, BC11_99_per, BC11_95_per, BC11_90_per, BC11_85_per, BC11_80_per, BC11_75_per],\n",
    "        'BC12':[BC12_total, BC12_99, BC12_95, BC12_90, BC12_85, BC12_80, BC12_75, BC12_99_per, BC12_95_per, BC12_90_per, BC12_85_per, BC12_80_per, BC12_75_per]\n",
    "       }\n",
    "\n",
    "\n",
    "accurate_rate = pd.DataFrame(data, index = ['Barcode total count', 'above 99%', 'above 95%', 'above 90%', 'above 85%', 'above 80%', 'above75%', \n",
    "                                            'above 99% percentage', 'above 95% percentage', 'above 90% percentage', 'above 85% percentage', 'above 80% percentage', 'above75% percentage'])\n",
    "\n",
    "#print('BC01 total count: ', len(raw_sucess_BC01))\n",
    "#rint(BC01_99, BC01_95, BC01_90, BC01_85, BC01_80, BC01_75)\n",
    "#rint(BC01_99_per, BC01_95_per, BC01_90_per, BC01_85_per, BC01_80_per, BC01_75_per)\n",
    "\n",
    "accurate_rate.to_csv(result_path +'accurate_rate_barcode_1_12.csv', encoding='utf-8')\n",
    "accurate_rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>read_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxid</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       read_count\n",
       "taxid            \n",
       "2            3436\n",
       "29              1\n",
       "39              1\n",
       "54             10\n",
       "356            26"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#####################\n",
    "# BC01 groupby taxid\n",
    "\n",
    "taxid_list_bc01 = pd.DataFrame(raw_sucess_BC01.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc01.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc01.to_csv(result_path + 'taxid_list_bc01.csv', encoding='utf-8')\n",
    "taxid_list_bc01.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC02 groupby taxid\n",
    "\n",
    "taxid_list_bc02 = pd.DataFrame(raw_sucess_BC02.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc02.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc02.to_csv(result_path + 'taxid_list_bc02.csv', encoding='utf-8')\n",
    "taxid_list_bc02.head()\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC03 groupby taxid\n",
    "\n",
    "taxid_list_bc03 = pd.DataFrame(raw_sucess_BC03.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc03.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc03.to_csv(result_path + 'taxid_list_bc03.csv', encoding='utf-8')\n",
    "taxid_list_bc03.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC04 groupby taxid\n",
    "\n",
    "taxid_list_bc04 = pd.DataFrame(raw_sucess_BC04.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc04.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc04.to_csv(result_path + 'taxid_list_bc04.csv', encoding='utf-8')\n",
    "taxid_list_bc04.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC05 groupby taxid\n",
    "\n",
    "taxid_list_bc05 = pd.DataFrame(raw_sucess_BC05.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc05.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc05.to_csv(result_path + 'taxid_list_bc05.csv', encoding='utf-8')\n",
    "taxid_list_bc05.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC06 groupby taxid\n",
    "\n",
    "taxid_list_bc06 = pd.DataFrame(raw_sucess_BC06.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc06.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc06.to_csv(result_path + 'taxid_list_bc06.csv', encoding='utf-8')\n",
    "taxid_list_bc06.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC07 groupby taxid\n",
    "\n",
    "taxid_list_bc07 = pd.DataFrame(raw_sucess_BC07.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc07.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc07.to_csv(result_path + 'taxid_list_bc07.csv', encoding='utf-8')\n",
    "taxid_list_bc07.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC08 groupby taxid\n",
    "\n",
    "taxid_list_bc08 = pd.DataFrame(raw_sucess_BC08.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc08.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc08.to_csv(result_path + 'taxid_list_bc08.csv', encoding='utf-8')\n",
    "taxid_list_bc08.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC09 groupby taxid\n",
    "\n",
    "taxid_list_bc09 = pd.DataFrame(raw_sucess_BC09.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc09.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc09.to_csv(result_path + 'taxid_list_bc09.csv', encoding='utf-8')\n",
    "taxid_list_bc09.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC10 groupby taxid\n",
    "\n",
    "taxid_list_bc10 = pd.DataFrame(raw_sucess_BC10.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc10.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc10.to_csv(result_path + 'taxid_list_bc10.csv', encoding='utf-8')\n",
    "taxid_list_bc10.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC11 groupby taxid\n",
    "\n",
    "taxid_list_bc11 = pd.DataFrame(raw_sucess_BC11.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc11.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc11.to_csv(result_path + 'taxid_list_bc11.csv', encoding='utf-8')\n",
    "taxid_list_bc11.head()\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# BC12 groupby taxid\n",
    "\n",
    "taxid_list_bc12 = pd.DataFrame(raw_sucess_BC12.groupby('taxid').count()['read_id'])\n",
    "\n",
    "taxid_list_bc12.columns = ['read_count']\n",
    "\n",
    "taxid_list_bc12.to_csv(result_path + 'taxid_list_bc12.csv', encoding='utf-8')\n",
    "taxid_list_bc12.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 2: Analysis 16s"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### No Selection in accuracy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/tcigene/anaconda3/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py:240: UserWarning: taxid 1432309 was translated into 1335613\n",
      "  warnings.warn(\"taxid %s was translated into %s\" %(taxid, merged_conversion[taxid]))\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxid</th>\n",
       "      <th>no rank</th>\n",
       "      <th>superkingdom</th>\n",
       "      <th>phylum</th>\n",
       "      <th>subphylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>suborder</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>species group</th>\n",
       "      <th>subspecies</th>\n",
       "      <th>subclass</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>131567</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>29</td>\n",
       "      <td>131567</td>\n",
       "      <td>2</td>\n",
       "      <td>1224</td>\n",
       "      <td>68525</td>\n",
       "      <td>28221</td>\n",
       "      <td>29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>39</td>\n",
       "      <td>131567</td>\n",
       "      <td>2</td>\n",
       "      <td>1224</td>\n",
       "      <td>68525</td>\n",
       "      <td>28221</td>\n",
       "      <td>29</td>\n",
       "      <td>80811</td>\n",
       "      <td>39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>54</td>\n",
       "      <td>131567</td>\n",
       "      <td>2</td>\n",
       "      <td>1224</td>\n",
       "      <td>68525</td>\n",
       "      <td>28221</td>\n",
       "      <td>29</td>\n",
       "      <td>224462</td>\n",
       "      <td>224463</td>\n",
       "      <td>53</td>\n",
       "      <td>54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>356</td>\n",
       "      <td>131567</td>\n",
       "      <td>2</td>\n",
       "      <td>1224</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28211</td>\n",
       "      <td>356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   taxid no rank superkingdom phylum subphylum  class order suborder  family  \\\n",
       "0      2  131567            2    NaN       NaN    NaN   NaN      NaN     NaN   \n",
       "1     29  131567            2   1224     68525  28221    29      NaN     NaN   \n",
       "2     39  131567            2   1224     68525  28221    29    80811      39   \n",
       "3     54  131567            2   1224     68525  28221    29   224462  224463   \n",
       "4    356  131567            2   1224       NaN  28211   356      NaN     NaN   \n",
       "\n",
       "  genus species species group subspecies subclass  \n",
       "0   NaN     NaN           NaN        NaN      NaN  \n",
       "1   NaN     NaN           NaN        NaN      NaN  \n",
       "2   NaN     NaN           NaN        NaN      NaN  \n",
       "3    53      54           NaN        NaN      NaN  \n",
       "4   NaN     NaN           NaN        NaN      NaN  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##############\n",
    "# BC01 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid01 = list(taxid_list_bc01.index)\n",
    "read_count01 = list(taxid_list_bc01['read_count'])\n",
    "total_count01 = len(raw_sucess_BC01)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage01 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage01['taxid'] = taxid01\n",
    "id_lineage01.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid01)):\n",
    "    L = ncbi.get_lineage(taxid01[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage01.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage01.head()\n",
    "      \n",
    "    \n",
    "    \n",
    "##############\n",
    "# BC02 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid02 = list(taxid_list_bc02.index)\n",
    "read_count02 = list(taxid_list_bc02['read_count'])\n",
    "total_count02 = len(raw_sucess_BC02)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage02 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage02['taxid'] = taxid02\n",
    "id_lineage02.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid02)):\n",
    "    L = ncbi.get_lineage(taxid02[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage02.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage02.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC03 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid03 = list(taxid_list_bc03.index)\n",
    "read_count03 = list(taxid_list_bc03['read_count'])\n",
    "total_count03 = len(raw_sucess_BC03)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage03 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage03['taxid'] = taxid03\n",
    "id_lineage03.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid03)):\n",
    "    L = ncbi.get_lineage(taxid03[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage03.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage03.head()\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC04 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid04 = list(taxid_list_bc04.index)\n",
    "read_count04 = list(taxid_list_bc04['read_count'])\n",
    "total_count04 = len(raw_sucess_BC04)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage04 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage04['taxid'] = taxid04\n",
    "id_lineage04.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid04)):\n",
    "    L = ncbi.get_lineage(taxid04[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage04.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage04.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC05 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid05 = list(taxid_list_bc05.index)\n",
    "read_count05 = list(taxid_list_bc05['read_count'])\n",
    "total_count05 = len(raw_sucess_BC05)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage05 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage05['taxid'] = taxid05\n",
    "id_lineage05.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid05)):\n",
    "    L = ncbi.get_lineage(taxid05[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage05.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage05.head()\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC06 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid06 = list(taxid_list_bc06.index)\n",
    "read_count06 = list(taxid_list_bc06['read_count'])\n",
    "total_count06 = len(raw_sucess_BC06)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage06 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage06['taxid'] = taxid06\n",
    "id_lineage06.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid06)):\n",
    "    L = ncbi.get_lineage(taxid06[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage06.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage06.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC07 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid07 = list(taxid_list_bc07.index)\n",
    "read_count07 = list(taxid_list_bc07['read_count'])\n",
    "total_count07 = len(raw_sucess_BC07)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage07 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage07['taxid'] = taxid07\n",
    "id_lineage07.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid07)):\n",
    "    L = ncbi.get_lineage(taxid07[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage07.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage07.head()\n",
    "      \n",
    "    \n",
    "    \n",
    "##############\n",
    "# BC08 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid08 = list(taxid_list_bc08.index)\n",
    "read_count08 = list(taxid_list_bc08['read_count'])\n",
    "total_count08 = len(raw_sucess_BC08)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage08 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage08['taxid'] = taxid08\n",
    "id_lineage08.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid08)):\n",
    "    L = ncbi.get_lineage(taxid08[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage08.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage08.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC09 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid09 = list(taxid_list_bc09.index)\n",
    "read_count09 = list(taxid_list_bc09['read_count'])\n",
    "total_count09 = len(raw_sucess_BC09)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage09 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage09['taxid'] = taxid09\n",
    "id_lineage09.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid09)):\n",
    "    L = ncbi.get_lineage(taxid09[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage09.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage09.head()\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC10 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid10 = list(taxid_list_bc10.index)\n",
    "read_count10 = list(taxid_list_bc10['read_count'])\n",
    "total_count10 = len(raw_sucess_BC10)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage10 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage10['taxid'] = taxid10\n",
    "id_lineage10.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid10)):\n",
    "    L = ncbi.get_lineage(taxid10[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage10.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage10.head()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC11 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid11 = list(taxid_list_bc11.index)\n",
    "read_count11 = list(taxid_list_bc11['read_count'])\n",
    "total_count11 = len(raw_sucess_BC11)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage11 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage11['taxid'] = taxid11\n",
    "id_lineage11.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid11)):\n",
    "    L = ncbi.get_lineage(taxid11[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage11.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage11.head()\n",
    "\n",
    "\n",
    "\n",
    "##############\n",
    "# BC12 lineage\n",
    "##############\n",
    "\n",
    "############\n",
    "# get taxid \n",
    "\n",
    "taxid12 = list(taxid_list_bc12.index)\n",
    "read_count12 = list(taxid_list_bc12['read_count'])\n",
    "total_count12 = len(raw_sucess_BC12)\n",
    "\n",
    "\n",
    "##########################################################\n",
    "# Create id_lineage dataframe to match the id with lineage\n",
    "\n",
    "id_lineage12 = pd.DataFrame(columns= ['taxid'])\n",
    "id_lineage12['taxid'] = taxid12\n",
    "id_lineage12.head()\n",
    "\n",
    "###############################################\n",
    "# the lineage of id add to id_lineage dataframe\n",
    "\n",
    "for i in range(len(taxid12)):\n",
    "    L = ncbi.get_lineage(taxid12[i])\n",
    "    \n",
    "    for node in L:\n",
    "        \n",
    "        rank = ncbi.get_rank([node])\n",
    "        id_lineage12.at[i, list(rank.values())[0]] = str(node)\n",
    "        \n",
    "\n",
    "id_lineage12.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>147803</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           name  read_count  percentage\n",
       "0      Bacteria      147803       100.0\n",
       "1  Unclassified           0         0.0"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###############################\n",
    "# BC01 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid01, id_lineage01['superkingdom'], read_count01)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC01_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC02 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid02, id_lineage02['superkingdom'], read_count02)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC02_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC03 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid03, id_lineage03['superkingdom'], read_count03)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC03_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC04 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid04, id_lineage04['superkingdom'], read_count04)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC04_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC05 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid05, id_lineage05['superkingdom'], read_count05)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC05_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC06 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid06, id_lineage06['superkingdom'], read_count06)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC06_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC07 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid07, id_lineage07['superkingdom'], read_count07)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC07_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC08 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid08, id_lineage08['superkingdom'], read_count08)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC08_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC09 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid09, id_lineage09['superkingdom'], read_count09)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC09_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC10 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid10, id_lineage10['superkingdom'], read_count10)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC10_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC11 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid11, id_lineage11['superkingdom'], read_count11)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC11_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom\n",
    "\n",
    "\n",
    "###############################\n",
    "# BC012 superkingdom: Read count \n",
    "\n",
    "superkingdom = pd.DataFrame(list(zip(taxid12, id_lineage12['superkingdom'], read_count12)), \n",
    "                            columns=['taxid', 'superkingdom', 'read_count'] )\n",
    "\n",
    "\n",
    "count_superkingdom = pd.DataFrame(superkingdom.groupby('superkingdom')['read_count'].sum())\n",
    "count_superkingdom.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_superkingdom.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_superkingdom['name'] = name\n",
    "count_superkingdom = count_superkingdom.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_superkingdom['percentage'] = (count_superkingdom['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_superkingdom.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_superkingdom.sum(axis = 0, skipna = True)[2]\n",
    "count_superkingdom = count_superkingdom.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_superkingdom.to_csv(result_path + 'BC12_count_superkingdom_noSelect.csv', encoding='utf-8')\n",
    "count_superkingdom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>72409</td>\n",
       "      <td>48.990210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>65905</td>\n",
       "      <td>44.589758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>3748</td>\n",
       "      <td>2.535808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cyanobacteria</td>\n",
       "      <td>382</td>\n",
       "      <td>0.258452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tenericutes</td>\n",
       "      <td>87</td>\n",
       "      <td>0.058862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>48</td>\n",
       "      <td>0.032476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>25</td>\n",
       "      <td>0.016914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Lentisphaerae</td>\n",
       "      <td>11</td>\n",
       "      <td>0.007442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Candidatus Melainabacteria</td>\n",
       "      <td>5</td>\n",
       "      <td>0.003383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Synergistetes</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Chlorobi</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Spirochaetes</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Fusobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>5177</td>\n",
       "      <td>3.502635</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          name  read_count  percentage\n",
       "0                Bacteroidetes       72409   48.990210\n",
       "1                   Firmicutes       65905   44.589758\n",
       "2               Proteobacteria        3748    2.535808\n",
       "3                Cyanobacteria         382    0.258452\n",
       "4                  Tenericutes          87    0.058862\n",
       "5               Actinobacteria          48    0.032476\n",
       "6              Verrucomicrobia          25    0.016914\n",
       "7                Lentisphaerae          11    0.007442\n",
       "8   Candidatus Melainabacteria           5    0.003383\n",
       "9                Synergistetes           3    0.002030\n",
       "10                    Chlorobi           1    0.000677\n",
       "11                Spirochaetes           1    0.000677\n",
       "12                Fusobacteria           1    0.000677\n",
       "13                Unclassified        5177    3.502635"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#########################\n",
    "# BC01 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid01, id_lineage01['phylum'], read_count01)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC01_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC02 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid02, id_lineage02['phylum'], read_count02)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC02_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC03 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid03, id_lineage03['phylum'], read_count03)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC03_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC04 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid04, id_lineage04['phylum'], read_count04)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC04_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC05 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid05, id_lineage05['phylum'], read_count05)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC05_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC06 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid06, id_lineage06['phylum'], read_count06)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC06_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC07 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid07, id_lineage07['phylum'], read_count07)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC07_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC08 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid08, id_lineage08['phylum'], read_count08)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC08_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC09 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid09, id_lineage09['phylum'], read_count09)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC09_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC10 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid10, id_lineage10['phylum'], read_count10)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC10_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC11 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid11, id_lineage11['phylum'], read_count11)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC11_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC12 phylum: Read count \n",
    "\n",
    "phylum = pd.DataFrame(list(zip(taxid12, id_lineage12['phylum'], read_count12)), \n",
    "                            columns=['taxid', 'phylum', 'read_count'] )\n",
    "\n",
    "\n",
    "count_phylum = pd.DataFrame(phylum.groupby('phylum')['read_count'].sum())\n",
    "count_phylum.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_phylum.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_phylum['name'] = name\n",
    "count_phylum = count_phylum.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_phylum['percentage'] = (count_phylum['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_phylum.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_phylum.sum(axis = 0, skipna = True)[2]\n",
    "count_phylum = count_phylum.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_phylum.to_csv(result_path + 'BC12_count_phylum_noSelect.csv', encoding='utf-8')\n",
    "count_phylum\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>72382</td>\n",
       "      <td>48.971942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Clostridia</td>\n",
       "      <td>52202</td>\n",
       "      <td>35.318634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Negativicutes</td>\n",
       "      <td>8575</td>\n",
       "      <td>5.801641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>1468</td>\n",
       "      <td>0.993214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Betaproteobacteria</td>\n",
       "      <td>1424</td>\n",
       "      <td>0.963445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>506</td>\n",
       "      <td>0.342348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Erysipelotrichia</td>\n",
       "      <td>347</td>\n",
       "      <td>0.234772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Bacilli</td>\n",
       "      <td>320</td>\n",
       "      <td>0.216504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>297</td>\n",
       "      <td>0.200943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Mollicutes</td>\n",
       "      <td>87</td>\n",
       "      <td>0.058862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>25</td>\n",
       "      <td>0.016914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>23</td>\n",
       "      <td>0.015561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Tissierellia</td>\n",
       "      <td>18</td>\n",
       "      <td>0.012178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Coriobacteriia</td>\n",
       "      <td>16</td>\n",
       "      <td>0.010825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Lentisphaeria</td>\n",
       "      <td>9</td>\n",
       "      <td>0.006089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Synergistia</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Acidimicrobiia</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Flavobacteriia</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Fusobacteriia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Chlorobia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Epsilonproteobacteria</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>10092</td>\n",
       "      <td>6.828008</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     name  read_count  percentage\n",
       "0             Bacteroidia       72382   48.971942\n",
       "1              Clostridia       52202   35.318634\n",
       "2           Negativicutes        8575    5.801641\n",
       "3     Gammaproteobacteria        1468    0.993214\n",
       "4      Betaproteobacteria        1424    0.963445\n",
       "5     Alphaproteobacteria         506    0.342348\n",
       "6        Erysipelotrichia         347    0.234772\n",
       "7                 Bacilli         320    0.216504\n",
       "8     Deltaproteobacteria         297    0.200943\n",
       "9              Mollicutes          87    0.058862\n",
       "10       Verrucomicrobiae          25    0.016914\n",
       "11         Actinobacteria          23    0.015561\n",
       "12           Tissierellia          18    0.012178\n",
       "13         Coriobacteriia          16    0.010825\n",
       "14          Lentisphaeria           9    0.006089\n",
       "15            Synergistia           3    0.002030\n",
       "16         Acidimicrobiia           3    0.002030\n",
       "17         Flavobacteriia           2    0.001353\n",
       "18          Fusobacteriia           1    0.000677\n",
       "19           Spirochaetia           1    0.000677\n",
       "20              Chlorobia           1    0.000677\n",
       "21  Epsilonproteobacteria           1    0.000677\n",
       "22           Unclassified       10092    6.828008"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "########################\n",
    "# BC01 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid01, id_lineage01['class'], read_count01)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC01_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC02 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid02, id_lineage02['class'], read_count02)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC02_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC03 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid03, id_lineage03['class'], read_count03)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC03_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC04 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid04, id_lineage04['class'], read_count04)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC04_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC05 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid05, id_lineage05['class'], read_count05)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC05_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC06 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid06, id_lineage06['class'], read_count06)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC06_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC07 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid07, id_lineage07['class'], read_count07)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC07_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC08 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid08, id_lineage08['class'], read_count08)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC08_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC09 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid09, id_lineage09['class'], read_count09)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC09_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC10 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid10, id_lineage10['class'], read_count10)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC10_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC11 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid11, id_lineage11['class'], read_count11)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC11_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class\n",
    "\n",
    "\n",
    "########################\n",
    "# BC12 class: Read count \n",
    "\n",
    "class_ = pd.DataFrame(list(zip(taxid12, id_lineage12['class'], read_count12)), \n",
    "                            columns=['taxid', 'class', 'read_count'] )\n",
    "\n",
    "\n",
    "count_class = pd.DataFrame(class_.groupby('class')['read_count'].sum())\n",
    "count_class.insert(0, 'name', '')\n",
    "\n",
    "\n",
    "name = []\n",
    "for i in count_class.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_class['name'] = name\n",
    "count_class = count_class.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_class['percentage'] = (count_class['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_class.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_class.sum(axis = 0, skipna = True)[2]\n",
    "count_class = count_class.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_class.to_csv(result_path + 'BC12_count_class_noSelect.csv', encoding='utf-8')\n",
    "count_class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>72358</td>\n",
       "      <td>48.955705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Clostridiales</td>\n",
       "      <td>52168</td>\n",
       "      <td>35.295630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Acidaminococcales</td>\n",
       "      <td>3809</td>\n",
       "      <td>2.577079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Selenomonadales</td>\n",
       "      <td>2145</td>\n",
       "      <td>1.451256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>1406</td>\n",
       "      <td>0.951266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Enterobacterales</td>\n",
       "      <td>1309</td>\n",
       "      <td>0.885638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Erysipelotrichales</td>\n",
       "      <td>347</td>\n",
       "      <td>0.234772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Desulfovibrionales</td>\n",
       "      <td>265</td>\n",
       "      <td>0.179293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Lactobacillales</td>\n",
       "      <td>229</td>\n",
       "      <td>0.154936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Caulobacterales</td>\n",
       "      <td>135</td>\n",
       "      <td>0.091338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Pasteurellales</td>\n",
       "      <td>117</td>\n",
       "      <td>0.079159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Oscillatoriales</td>\n",
       "      <td>82</td>\n",
       "      <td>0.055479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Bacillales</td>\n",
       "      <td>72</td>\n",
       "      <td>0.048713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Veillonellales</td>\n",
       "      <td>71</td>\n",
       "      <td>0.048037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Rhizobiales</td>\n",
       "      <td>41</td>\n",
       "      <td>0.027740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Mycoplasmatales</td>\n",
       "      <td>40</td>\n",
       "      <td>0.027063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Entomoplasmatales</td>\n",
       "      <td>33</td>\n",
       "      <td>0.022327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Rhodospirillales</td>\n",
       "      <td>32</td>\n",
       "      <td>0.021650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Kiloniellales</td>\n",
       "      <td>29</td>\n",
       "      <td>0.019621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Verrucomicrobiales</td>\n",
       "      <td>25</td>\n",
       "      <td>0.016914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Myxococcales</td>\n",
       "      <td>25</td>\n",
       "      <td>0.016914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Rhodobacterales</td>\n",
       "      <td>24</td>\n",
       "      <td>0.016238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Nostocales</td>\n",
       "      <td>19</td>\n",
       "      <td>0.012855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Tissierellales</td>\n",
       "      <td>16</td>\n",
       "      <td>0.010825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Coriobacteriales</td>\n",
       "      <td>14</td>\n",
       "      <td>0.009472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Sphingomonadales</td>\n",
       "      <td>6</td>\n",
       "      <td>0.004059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Micromonosporales</td>\n",
       "      <td>6</td>\n",
       "      <td>0.004059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Synechococcales</td>\n",
       "      <td>6</td>\n",
       "      <td>0.004059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Kordiimonadales</td>\n",
       "      <td>5</td>\n",
       "      <td>0.003383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Vampirovibrionales</td>\n",
       "      <td>5</td>\n",
       "      <td>0.003383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Desulfuromonadales</td>\n",
       "      <td>5</td>\n",
       "      <td>0.003383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>4</td>\n",
       "      <td>0.002706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Streptomycetales</td>\n",
       "      <td>4</td>\n",
       "      <td>0.002706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Vibrionales</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Acidimicrobiales</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Synergistales</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Oceanospirillales</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Eggerthellales</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Flavobacteriales</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Marinilabiliales</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Xanthomonadales</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Chromatiales</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Orbales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Rickettsiales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Micrococcales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Alteromonadales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Thermoanaerobacterales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Kineosporiales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Spirochaetales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Sneathiellales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Aeromonadales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Campylobacterales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Methylococcales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Acholeplasmatales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Fusobacteriales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>Chlorobiales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Natranaerobiales</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>12914</td>\n",
       "      <td>8.737306</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      name  read_count  percentage\n",
       "0            Bacteroidales       72358   48.955705\n",
       "1            Clostridiales       52168   35.295630\n",
       "2        Acidaminococcales        3809    2.577079\n",
       "3          Selenomonadales        2145    1.451256\n",
       "4          Burkholderiales        1406    0.951266\n",
       "5         Enterobacterales        1309    0.885638\n",
       "6       Erysipelotrichales         347    0.234772\n",
       "7       Desulfovibrionales         265    0.179293\n",
       "8          Lactobacillales         229    0.154936\n",
       "9          Caulobacterales         135    0.091338\n",
       "10          Pasteurellales         117    0.079159\n",
       "11         Oscillatoriales          82    0.055479\n",
       "12              Bacillales          72    0.048713\n",
       "13          Veillonellales          71    0.048037\n",
       "14             Rhizobiales          41    0.027740\n",
       "15         Mycoplasmatales          40    0.027063\n",
       "16       Entomoplasmatales          33    0.022327\n",
       "17        Rhodospirillales          32    0.021650\n",
       "18           Kiloniellales          29    0.019621\n",
       "19      Verrucomicrobiales          25    0.016914\n",
       "20            Myxococcales          25    0.016914\n",
       "21         Rhodobacterales          24    0.016238\n",
       "22              Nostocales          19    0.012855\n",
       "23          Tissierellales          16    0.010825\n",
       "24        Coriobacteriales          14    0.009472\n",
       "25        Sphingomonadales           6    0.004059\n",
       "26       Micromonosporales           6    0.004059\n",
       "27         Synechococcales           6    0.004059\n",
       "28         Kordiimonadales           5    0.003383\n",
       "29      Vampirovibrionales           5    0.003383\n",
       "30      Desulfuromonadales           5    0.003383\n",
       "31         Pseudomonadales           4    0.002706\n",
       "32        Streptomycetales           4    0.002706\n",
       "33             Vibrionales           3    0.002030\n",
       "34        Acidimicrobiales           3    0.002030\n",
       "35           Synergistales           3    0.002030\n",
       "36       Oceanospirillales           2    0.001353\n",
       "37          Eggerthellales           2    0.001353\n",
       "38        Flavobacteriales           2    0.001353\n",
       "39        Marinilabiliales           2    0.001353\n",
       "40         Xanthomonadales           2    0.001353\n",
       "41            Chromatiales           2    0.001353\n",
       "42     Propionibacteriales           1    0.000677\n",
       "43                 Orbales           1    0.000677\n",
       "44           Rickettsiales           1    0.000677\n",
       "45           Micrococcales           1    0.000677\n",
       "46         Alteromonadales           1    0.000677\n",
       "47  Thermoanaerobacterales           1    0.000677\n",
       "48          Kineosporiales           1    0.000677\n",
       "49          Spirochaetales           1    0.000677\n",
       "50          Sneathiellales           1    0.000677\n",
       "51           Aeromonadales           1    0.000677\n",
       "52       Campylobacterales           1    0.000677\n",
       "53         Methylococcales           1    0.000677\n",
       "54       Acholeplasmatales           1    0.000677\n",
       "55         Fusobacteriales           1    0.000677\n",
       "56            Chlorobiales           1    0.000677\n",
       "57        Natranaerobiales           1    0.000677\n",
       "58            Unclassified       12914    8.737306"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "########################\n",
    "# BC01 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid01, id_lineage01['order'], read_count01)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC01_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC02 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid02, id_lineage02['order'], read_count02)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC02_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC03 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid03, id_lineage03['order'], read_count03)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC03_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC04 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid04, id_lineage04['order'], read_count04)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC04_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC05 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid05, id_lineage05['order'], read_count05)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC05_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC06 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid06, id_lineage06['order'], read_count06)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC06_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC07 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid07, id_lineage07['order'], read_count07)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC07_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC08 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid08, id_lineage08['order'], read_count08)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC08_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC09 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid09, id_lineage09['order'], read_count09)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC09_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC10 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid10, id_lineage10['order'], read_count10)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC10_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC11 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid11, id_lineage11['order'], read_count11)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC11_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order\n",
    "\n",
    "\n",
    "########################\n",
    "# BC12 order: Read count \n",
    "\n",
    "order = pd.DataFrame(list(zip(taxid12, id_lineage12['order'], read_count12)), \n",
    "                            columns=['taxid', 'order', 'read_count'] )\n",
    "\n",
    "\n",
    "count_order = pd.DataFrame(order.groupby('order')['read_count'].sum())\n",
    "count_order.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_order.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_order['name'] = name\n",
    "count_order = count_order.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_order['percentage'] = (count_order['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_order.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_order.sum(axis = 0, skipna = True)[2]\n",
    "count_order = count_order.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_order.to_csv(result_path + 'BC12_count_order_noSelect.csv', encoding='utf-8')\n",
    "count_order"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Prevotellaceae</td>\n",
       "      <td>26694</td>\n",
       "      <td>18.060527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tannerellaceae</td>\n",
       "      <td>20746</td>\n",
       "      <td>14.036251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>18992</td>\n",
       "      <td>12.849536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>17693</td>\n",
       "      <td>11.970664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>15446</td>\n",
       "      <td>10.450397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Acidaminococcaceae</td>\n",
       "      <td>3809</td>\n",
       "      <td>2.577079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>3743</td>\n",
       "      <td>2.532425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Selenomonadaceae</td>\n",
       "      <td>2009</td>\n",
       "      <td>1.359242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Rikenellaceae</td>\n",
       "      <td>1995</td>\n",
       "      <td>1.349770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Hungateiclostridiaceae</td>\n",
       "      <td>1254</td>\n",
       "      <td>0.848427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Enterobacteriaceae</td>\n",
       "      <td>1053</td>\n",
       "      <td>0.712435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Odoribacteraceae</td>\n",
       "      <td>915</td>\n",
       "      <td>0.619067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Sutterellaceae</td>\n",
       "      <td>821</td>\n",
       "      <td>0.555469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Erysipelotrichaceae</td>\n",
       "      <td>347</td>\n",
       "      <td>0.234772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Desulfovibrionaceae</td>\n",
       "      <td>264</td>\n",
       "      <td>0.178616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Caulobacteraceae</td>\n",
       "      <td>135</td>\n",
       "      <td>0.091338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Clostridiaceae</td>\n",
       "      <td>124</td>\n",
       "      <td>0.083895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Pasteurellaceae</td>\n",
       "      <td>117</td>\n",
       "      <td>0.079159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Streptococcaceae</td>\n",
       "      <td>115</td>\n",
       "      <td>0.077806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Barnesiellaceae</td>\n",
       "      <td>90</td>\n",
       "      <td>0.060892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Veillonellaceae</td>\n",
       "      <td>71</td>\n",
       "      <td>0.048037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Oxalobacteraceae</td>\n",
       "      <td>69</td>\n",
       "      <td>0.046684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Clostridiales Family XIII. Incertae Sedis</td>\n",
       "      <td>54</td>\n",
       "      <td>0.036535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Peptostreptococcaceae</td>\n",
       "      <td>51</td>\n",
       "      <td>0.034505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Gracilibacteraceae</td>\n",
       "      <td>47</td>\n",
       "      <td>0.031799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Christensenellaceae</td>\n",
       "      <td>45</td>\n",
       "      <td>0.030446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Sporomusaceae</td>\n",
       "      <td>43</td>\n",
       "      <td>0.029093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>41</td>\n",
       "      <td>0.027740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Mycoplasmataceae</td>\n",
       "      <td>40</td>\n",
       "      <td>0.027063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Lactobacillaceae</td>\n",
       "      <td>39</td>\n",
       "      <td>0.026386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Brucellaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>Kineosporiaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>Microbacteriaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Orbaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>Aeromonadaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Syntrophomonadaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Methylococcaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>Marinilabiliaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Spirochaetaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Alicyclobacillaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>Aerococcaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Synechococcaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Erwiniaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Morganellaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Chlorobiaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Thermoanaerobacteraceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Verrucomicrobiaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Crocinitomicaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Desulfuromonadaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Acholeplasmataceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Pseudoalteromonadaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>Heliobacteriaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Archangiaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Natranaerobiaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Flavobacteriaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Sneathiellaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Proteinivoraceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Prolixibacteraceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>Leptotrichiaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>30494</td>\n",
       "      <td>20.631516</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>112 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          name  read_count  percentage\n",
       "0                               Prevotellaceae       26694   18.060527\n",
       "1                               Tannerellaceae       20746   14.036251\n",
       "2                              Ruminococcaceae       18992   12.849536\n",
       "3                               Bacteroidaceae       17693   11.970664\n",
       "4                              Lachnospiraceae       15446   10.450397\n",
       "5                           Acidaminococcaceae        3809    2.577079\n",
       "6                             Oscillospiraceae        3743    2.532425\n",
       "7                             Selenomonadaceae        2009    1.359242\n",
       "8                                Rikenellaceae        1995    1.349770\n",
       "9                       Hungateiclostridiaceae        1254    0.848427\n",
       "10                          Enterobacteriaceae        1053    0.712435\n",
       "11                            Odoribacteraceae         915    0.619067\n",
       "12                              Sutterellaceae         821    0.555469\n",
       "13                         Erysipelotrichaceae         347    0.234772\n",
       "14                         Desulfovibrionaceae         264    0.178616\n",
       "15                            Caulobacteraceae         135    0.091338\n",
       "16                              Clostridiaceae         124    0.083895\n",
       "17                             Pasteurellaceae         117    0.079159\n",
       "18                            Streptococcaceae         115    0.077806\n",
       "19                             Barnesiellaceae          90    0.060892\n",
       "20                             Veillonellaceae          71    0.048037\n",
       "21                            Oxalobacteraceae          69    0.046684\n",
       "22   Clostridiales Family XIII. Incertae Sedis          54    0.036535\n",
       "23                       Peptostreptococcaceae          51    0.034505\n",
       "24                          Gracilibacteraceae          47    0.031799\n",
       "25                         Christensenellaceae          45    0.030446\n",
       "26                               Sporomusaceae          43    0.029093\n",
       "27                            Paenibacillaceae          41    0.027740\n",
       "28                            Mycoplasmataceae          40    0.027063\n",
       "29                            Lactobacillaceae          39    0.026386\n",
       "..                                         ...         ...         ...\n",
       "82                                Brucellaceae           1    0.000677\n",
       "83                             Kineosporiaceae           1    0.000677\n",
       "84                           Microbacteriaceae           1    0.000677\n",
       "85                                    Orbaceae           1    0.000677\n",
       "86                              Aeromonadaceae           1    0.000677\n",
       "87                         Syntrophomonadaceae           1    0.000677\n",
       "88                            Methylococcaceae           1    0.000677\n",
       "89                           Marinilabiliaceae           1    0.000677\n",
       "90                             Spirochaetaceae           1    0.000677\n",
       "91                         Alicyclobacillaceae           1    0.000677\n",
       "92                               Aerococcaceae           1    0.000677\n",
       "93                            Synechococcaceae           1    0.000677\n",
       "94                                 Erwiniaceae           1    0.000677\n",
       "95                              Morganellaceae           1    0.000677\n",
       "96                               Chlorobiaceae           1    0.000677\n",
       "97                     Thermoanaerobacteraceae           1    0.000677\n",
       "98                         Verrucomicrobiaceae           1    0.000677\n",
       "99                           Crocinitomicaceae           1    0.000677\n",
       "100                        Desulfuromonadaceae           1    0.000677\n",
       "101                         Acholeplasmataceae           1    0.000677\n",
       "102                     Pseudoalteromonadaceae           1    0.000677\n",
       "103                          Heliobacteriaceae           1    0.000677\n",
       "104                              Archangiaceae           1    0.000677\n",
       "105                          Natranaerobiaceae           1    0.000677\n",
       "106                          Flavobacteriaceae           1    0.000677\n",
       "107                            Sneathiellaceae           1    0.000677\n",
       "108                           Proteinivoraceae           1    0.000677\n",
       "109                         Prolixibacteraceae           1    0.000677\n",
       "110                           Leptotrichiaceae           1    0.000677\n",
       "111                               Unclassified       30494   20.631516\n",
       "\n",
       "[112 rows x 3 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#########################\n",
    "# BC01 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid01, id_lineage01['family'], read_count01)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC01_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC02 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid02, id_lineage02['family'], read_count02)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC02_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC03 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid03, id_lineage03['family'], read_count03)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC03_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC04 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid04, id_lineage04['family'], read_count04)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC04_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC05 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid05, id_lineage05['family'], read_count05)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC05_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC06 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid06, id_lineage06['family'], read_count06)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "\n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC06_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC07 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid07, id_lineage07['family'], read_count07)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC07_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC08 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid08, id_lineage08['family'], read_count08)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC08_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC09 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid09, id_lineage09['family'], read_count09)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC09_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC10 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid10, id_lineage10['family'], read_count10)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "\n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC10_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC11 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid11, id_lineage11['family'], read_count11)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC11_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family\n",
    "\n",
    "\n",
    "#########################\n",
    "# BC12 family: Read count \n",
    "\n",
    "family = pd.DataFrame(list(zip(taxid12, id_lineage12['family'], read_count12)), \n",
    "                            columns=['taxid', 'family', 'read_count'] )\n",
    "\n",
    "\n",
    "count_family = pd.DataFrame(family.groupby('family')['read_count'].sum())\n",
    "count_family.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_family.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_family['name'] = name\n",
    "count_family = count_family.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_family['percentage'] = (count_family['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_family.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_family.sum(axis = 0, skipna = True)[2]\n",
    "count_family = count_family.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_family.to_csv(result_path + 'BC12_count_family_noSelect.csv', encoding='utf-8')\n",
    "count_family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Prevotella</td>\n",
       "      <td>25886</td>\n",
       "      <td>17.513853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Parabacteroides</td>\n",
       "      <td>20456</td>\n",
       "      <td>13.840044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>17567</td>\n",
       "      <td>11.885415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Lachnoclostridium</td>\n",
       "      <td>4092</td>\n",
       "      <td>2.768550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Oscillibacter</td>\n",
       "      <td>3743</td>\n",
       "      <td>2.532425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Ruminococcus</td>\n",
       "      <td>2058</td>\n",
       "      <td>1.392394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Alistipes</td>\n",
       "      <td>1993</td>\n",
       "      <td>1.348416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Megamonas</td>\n",
       "      <td>1662</td>\n",
       "      <td>1.124470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Roseburia</td>\n",
       "      <td>1346</td>\n",
       "      <td>0.910672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Blautia</td>\n",
       "      <td>1026</td>\n",
       "      <td>0.694167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Klebsiella</td>\n",
       "      <td>625</td>\n",
       "      <td>0.422860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Anaerotignum</td>\n",
       "      <td>582</td>\n",
       "      <td>0.393767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Odoribacter</td>\n",
       "      <td>563</td>\n",
       "      <td>0.380912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Kineothrix</td>\n",
       "      <td>395</td>\n",
       "      <td>0.267248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Butyricimonas</td>\n",
       "      <td>343</td>\n",
       "      <td>0.232066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Paraprevotella</td>\n",
       "      <td>294</td>\n",
       "      <td>0.198913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Hungateiclostridium</td>\n",
       "      <td>257</td>\n",
       "      <td>0.173880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Desulfovibrio</td>\n",
       "      <td>249</td>\n",
       "      <td>0.168467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Ruminiclostridium</td>\n",
       "      <td>194</td>\n",
       "      <td>0.131256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Anaerostipes</td>\n",
       "      <td>150</td>\n",
       "      <td>0.101486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Brevundimonas</td>\n",
       "      <td>131</td>\n",
       "      <td>0.088631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Faecalicatena</td>\n",
       "      <td>124</td>\n",
       "      <td>0.083895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Citrobacter</td>\n",
       "      <td>116</td>\n",
       "      <td>0.078483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Haemophilus</td>\n",
       "      <td>91</td>\n",
       "      <td>0.061568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Barnesiella</td>\n",
       "      <td>90</td>\n",
       "      <td>0.060892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Clostridium</td>\n",
       "      <td>61</td>\n",
       "      <td>0.041271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Streptococcus</td>\n",
       "      <td>57</td>\n",
       "      <td>0.038565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Lactococcus</td>\n",
       "      <td>53</td>\n",
       "      <td>0.035859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Gracilibacter</td>\n",
       "      <td>47</td>\n",
       "      <td>0.031799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Christensenella</td>\n",
       "      <td>45</td>\n",
       "      <td>0.030446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>Photorhabdus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>Brasilonema</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>Halolactibacillus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>Desulfitobacterium</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>Burkholderia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>Salimicrobium</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>Desulfotomaculum</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>Rhodovulum</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>Neorickettsia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>Azospirillum</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>Geobacter</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>Carboxydothermus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>Hirschia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>Desulfofundulus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>Erwinia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>Brevibacillus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>Anaerospora</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>Maribacter</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>Pseudoalteromonas</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>Planococcus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>Ochrobactrum</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>Sneathiella</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>Kineosporia</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>Ornithinibacillus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>Marinobacterium</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>Oribacterium</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>Acinetobacter</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>Bacillus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>Desulfallas</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>62743</td>\n",
       "      <td>42.450424</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>181 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    name  read_count  percentage\n",
       "0             Prevotella       25886   17.513853\n",
       "1        Parabacteroides       20456   13.840044\n",
       "2            Bacteroides       17567   11.885415\n",
       "3      Lachnoclostridium        4092    2.768550\n",
       "4          Oscillibacter        3743    2.532425\n",
       "5           Ruminococcus        2058    1.392394\n",
       "6              Alistipes        1993    1.348416\n",
       "7              Megamonas        1662    1.124470\n",
       "8              Roseburia        1346    0.910672\n",
       "9                Blautia        1026    0.694167\n",
       "10            Klebsiella         625    0.422860\n",
       "11          Anaerotignum         582    0.393767\n",
       "12           Odoribacter         563    0.380912\n",
       "13            Kineothrix         395    0.267248\n",
       "14         Butyricimonas         343    0.232066\n",
       "15        Paraprevotella         294    0.198913\n",
       "16   Hungateiclostridium         257    0.173880\n",
       "17         Desulfovibrio         249    0.168467\n",
       "18     Ruminiclostridium         194    0.131256\n",
       "19          Anaerostipes         150    0.101486\n",
       "20         Brevundimonas         131    0.088631\n",
       "21         Faecalicatena         124    0.083895\n",
       "22           Citrobacter         116    0.078483\n",
       "23           Haemophilus          91    0.061568\n",
       "24           Barnesiella          90    0.060892\n",
       "25           Clostridium          61    0.041271\n",
       "26         Streptococcus          57    0.038565\n",
       "27           Lactococcus          53    0.035859\n",
       "28         Gracilibacter          47    0.031799\n",
       "29       Christensenella          45    0.030446\n",
       "..                   ...         ...         ...\n",
       "151         Photorhabdus           1    0.000677\n",
       "152          Brasilonema           1    0.000677\n",
       "153    Halolactibacillus           1    0.000677\n",
       "154   Desulfitobacterium           1    0.000677\n",
       "155         Burkholderia           1    0.000677\n",
       "156        Salimicrobium           1    0.000677\n",
       "157     Desulfotomaculum           1    0.000677\n",
       "158           Rhodovulum           1    0.000677\n",
       "159        Neorickettsia           1    0.000677\n",
       "160         Azospirillum           1    0.000677\n",
       "161            Geobacter           1    0.000677\n",
       "162     Carboxydothermus           1    0.000677\n",
       "163             Hirschia           1    0.000677\n",
       "164      Desulfofundulus           1    0.000677\n",
       "165              Erwinia           1    0.000677\n",
       "166        Brevibacillus           1    0.000677\n",
       "167          Anaerospora           1    0.000677\n",
       "168           Maribacter           1    0.000677\n",
       "169    Pseudoalteromonas           1    0.000677\n",
       "170          Planococcus           1    0.000677\n",
       "171         Ochrobactrum           1    0.000677\n",
       "172          Sneathiella           1    0.000677\n",
       "173          Kineosporia           1    0.000677\n",
       "174    Ornithinibacillus           1    0.000677\n",
       "175      Marinobacterium           1    0.000677\n",
       "176         Oribacterium           1    0.000677\n",
       "177        Acinetobacter           1    0.000677\n",
       "178             Bacillus           1    0.000677\n",
       "179          Desulfallas           1    0.000677\n",
       "180         Unclassified       62743   42.450424\n",
       "\n",
       "[181 rows x 3 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "########################\n",
    "# BC01 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid01, id_lineage01['genus'], read_count01)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC01_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC02 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid02, id_lineage02['genus'], read_count02)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC02_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC03 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid03, id_lineage03['genus'], read_count03)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC03_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC04 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid04, id_lineage04['genus'], read_count04)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC04_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "\n",
    "########################\n",
    "# BC05 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid05, id_lineage05['genus'], read_count05)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC05_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC06 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid06, id_lineage06['genus'], read_count06)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC06_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC07 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid07, id_lineage07['genus'], read_count07)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC07_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC08 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid08, id_lineage08['genus'], read_count08)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC08_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC09 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid09, id_lineage09['genus'], read_count09)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC09_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC10 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid10, id_lineage10['genus'], read_count10)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC10_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC11 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid11, id_lineage11['genus'], read_count11)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC11_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus\n",
    "\n",
    "\n",
    "########################\n",
    "# BC12 genus: Read count \n",
    "\n",
    "genus = pd.DataFrame(list(zip(taxid12, id_lineage12['genus'], read_count12)), \n",
    "                            columns=['taxid', 'genus', 'read_count'] )\n",
    "\n",
    "\n",
    "count_genus = pd.DataFrame(genus.groupby('genus')['read_count'].sum())\n",
    "count_genus.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_genus.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_genus['name'] = name\n",
    "count_genus = count_genus.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_genus['percentage'] = (count_genus['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_genus.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_genus.sum(axis = 0, skipna = True)[2]\n",
    "count_genus = count_genus.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_genus.to_csv(result_path + 'BC12_count_genus_noSelect.csv', encoding='utf-8')\n",
    "count_genus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>read_count</th>\n",
       "      <th>percentage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>25418</td>\n",
       "      <td>17.197215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Parabacteroides distasonis</td>\n",
       "      <td>16104</td>\n",
       "      <td>10.895584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Bacteroides plebeius</td>\n",
       "      <td>6189</td>\n",
       "      <td>4.187330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bacteroides coprocola</td>\n",
       "      <td>5497</td>\n",
       "      <td>3.719140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Parabacteroides merdae</td>\n",
       "      <td>3767</td>\n",
       "      <td>2.548663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Oscillibacter valericigenes</td>\n",
       "      <td>3031</td>\n",
       "      <td>2.050703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Ruminococcus albus</td>\n",
       "      <td>1916</td>\n",
       "      <td>1.296320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Bacteroides vulgatus</td>\n",
       "      <td>1682</td>\n",
       "      <td>1.138001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[Clostridium] bolteae</td>\n",
       "      <td>1356</td>\n",
       "      <td>0.917437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Alistipes putredinis</td>\n",
       "      <td>1318</td>\n",
       "      <td>0.891728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Megamonas rupellensis</td>\n",
       "      <td>1013</td>\n",
       "      <td>0.685372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Bacteroides massiliensis</td>\n",
       "      <td>959</td>\n",
       "      <td>0.648837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Bacteroides caccae</td>\n",
       "      <td>738</td>\n",
       "      <td>0.499313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Oscillibacter ruminantium</td>\n",
       "      <td>712</td>\n",
       "      <td>0.481722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Megamonas funiformis</td>\n",
       "      <td>601</td>\n",
       "      <td>0.406622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Bacteroides uniformis</td>\n",
       "      <td>590</td>\n",
       "      <td>0.399180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Odoribacter splanchnicus</td>\n",
       "      <td>563</td>\n",
       "      <td>0.380912</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>[Desulfotomaculum] guttoideum</td>\n",
       "      <td>547</td>\n",
       "      <td>0.370087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Roseburia faecis</td>\n",
       "      <td>543</td>\n",
       "      <td>0.367381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Anaerotignum lactatifermentans</td>\n",
       "      <td>542</td>\n",
       "      <td>0.366704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>541</td>\n",
       "      <td>0.366028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Parabacteroides goldsteinii</td>\n",
       "      <td>506</td>\n",
       "      <td>0.342348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Alistipes shahii</td>\n",
       "      <td>430</td>\n",
       "      <td>0.290928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>[Clostridium] indolis</td>\n",
       "      <td>422</td>\n",
       "      <td>0.285515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Kineothrix alysoides</td>\n",
       "      <td>395</td>\n",
       "      <td>0.267248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Roseburia hominis</td>\n",
       "      <td>337</td>\n",
       "      <td>0.228006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>[Clostridium] saccharolyticum</td>\n",
       "      <td>294</td>\n",
       "      <td>0.198913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Paraprevotella clara</td>\n",
       "      <td>293</td>\n",
       "      <td>0.198237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Lachnoclostridium pacaense</td>\n",
       "      <td>291</td>\n",
       "      <td>0.196884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Roseburia inulinivorans</td>\n",
       "      <td>272</td>\n",
       "      <td>0.184029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>Coprococcus comes</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>Paenibacillus provencensis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>Prevotella amnii</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>Anaerobranca horikoshii</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>Gilliamella mensalis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>Bhargavaea beijingensis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>Macrococcus bohemicus</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>Tissierella creatinini</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504</th>\n",
       "      <td>Prevotella micans</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505</th>\n",
       "      <td>Paenibacillus graminis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>506</th>\n",
       "      <td>Propionispira arcuata</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507</th>\n",
       "      <td>Nocardioides islandensis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>508</th>\n",
       "      <td>Brasilonema terrestre</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509</th>\n",
       "      <td>Prevotella oryzae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>Marinobacterium aestuarii</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>511</th>\n",
       "      <td>Acutalibacter muris</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>512</th>\n",
       "      <td>Paraprevotella xylaniphila</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>513</th>\n",
       "      <td>Lactobacillus coleohominis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>Maritalea myrionectae</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>515</th>\n",
       "      <td>Sutterella massiliensis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>516</th>\n",
       "      <td>Granulicatella adiacens</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>517</th>\n",
       "      <td>Phaeospirillum chandramohanii</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>Pectobacterium brasiliense</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>Atopostipes suicloacalis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>Delftia tsuruhatensis</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>Lachnoanaerobaculum saburreum</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>Clostridium chauvoei</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>Herbaspirillum autotrophicum</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>524</th>\n",
       "      <td>Raoultella ornithinolytica</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>525</th>\n",
       "      <td>Unclassified</td>\n",
       "      <td>62742</td>\n",
       "      <td>42.449747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>526 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               name  read_count  percentage\n",
       "0                  Prevotella copri       25418   17.197215\n",
       "1        Parabacteroides distasonis       16104   10.895584\n",
       "2              Bacteroides plebeius        6189    4.187330\n",
       "3             Bacteroides coprocola        5497    3.719140\n",
       "4            Parabacteroides merdae        3767    2.548663\n",
       "5       Oscillibacter valericigenes        3031    2.050703\n",
       "6                Ruminococcus albus        1916    1.296320\n",
       "7              Bacteroides vulgatus        1682    1.138001\n",
       "8             [Clostridium] bolteae        1356    0.917437\n",
       "9              Alistipes putredinis        1318    0.891728\n",
       "10            Megamonas rupellensis        1013    0.685372\n",
       "11         Bacteroides massiliensis         959    0.648837\n",
       "12               Bacteroides caccae         738    0.499313\n",
       "13        Oscillibacter ruminantium         712    0.481722\n",
       "14             Megamonas funiformis         601    0.406622\n",
       "15            Bacteroides uniformis         590    0.399180\n",
       "16         Odoribacter splanchnicus         563    0.380912\n",
       "17    [Desulfotomaculum] guttoideum         547    0.370087\n",
       "18                 Roseburia faecis         543    0.367381\n",
       "19   Anaerotignum lactatifermentans         542    0.366704\n",
       "20            Klebsiella pneumoniae         541    0.366028\n",
       "21      Parabacteroides goldsteinii         506    0.342348\n",
       "22                 Alistipes shahii         430    0.290928\n",
       "23            [Clostridium] indolis         422    0.285515\n",
       "24             Kineothrix alysoides         395    0.267248\n",
       "25                Roseburia hominis         337    0.228006\n",
       "26    [Clostridium] saccharolyticum         294    0.198913\n",
       "27             Paraprevotella clara         293    0.198237\n",
       "28       Lachnoclostridium pacaense         291    0.196884\n",
       "29          Roseburia inulinivorans         272    0.184029\n",
       "..                              ...         ...         ...\n",
       "496               Coprococcus comes           1    0.000677\n",
       "497      Paenibacillus provencensis           1    0.000677\n",
       "498                Prevotella amnii           1    0.000677\n",
       "499         Anaerobranca horikoshii           1    0.000677\n",
       "500            Gilliamella mensalis           1    0.000677\n",
       "501         Bhargavaea beijingensis           1    0.000677\n",
       "502           Macrococcus bohemicus           1    0.000677\n",
       "503          Tissierella creatinini           1    0.000677\n",
       "504               Prevotella micans           1    0.000677\n",
       "505          Paenibacillus graminis           1    0.000677\n",
       "506           Propionispira arcuata           1    0.000677\n",
       "507        Nocardioides islandensis           1    0.000677\n",
       "508           Brasilonema terrestre           1    0.000677\n",
       "509               Prevotella oryzae           1    0.000677\n",
       "510       Marinobacterium aestuarii           1    0.000677\n",
       "511             Acutalibacter muris           1    0.000677\n",
       "512      Paraprevotella xylaniphila           1    0.000677\n",
       "513      Lactobacillus coleohominis           1    0.000677\n",
       "514           Maritalea myrionectae           1    0.000677\n",
       "515         Sutterella massiliensis           1    0.000677\n",
       "516         Granulicatella adiacens           1    0.000677\n",
       "517   Phaeospirillum chandramohanii           1    0.000677\n",
       "518      Pectobacterium brasiliense           1    0.000677\n",
       "519        Atopostipes suicloacalis           1    0.000677\n",
       "520           Delftia tsuruhatensis           1    0.000677\n",
       "521   Lachnoanaerobaculum saburreum           1    0.000677\n",
       "522            Clostridium chauvoei           1    0.000677\n",
       "523    Herbaspirillum autotrophicum           1    0.000677\n",
       "524      Raoultella ornithinolytica           1    0.000677\n",
       "525                    Unclassified       62742   42.449747\n",
       "\n",
       "[526 rows x 3 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##########################\n",
    "# BC01 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid01, id_lineage01['species'], read_count01)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count01)*100\n",
    "\n",
    "unclassified_count = total_count01 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC01_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC02 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid02, id_lineage02['species'], read_count02)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count02)*100\n",
    "\n",
    "unclassified_count = total_count02 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC02_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC03 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid03, id_lineage03['species'], read_count03)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count03)*100\n",
    "\n",
    "unclassified_count = total_count03 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC03_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC04 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid04, id_lineage04['species'], read_count04)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count04)*100\n",
    "\n",
    "unclassified_count = total_count04 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC04_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC05 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid05, id_lineage05['species'], read_count05)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count05)*100\n",
    "\n",
    "unclassified_count = total_count05 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC05_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC06 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid06, id_lineage06['species'], read_count06)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count06)*100\n",
    "\n",
    "unclassified_count = total_count06 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC06_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC07 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid07, id_lineage07['species'], read_count07)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count07)*100\n",
    "\n",
    "unclassified_count = total_count07 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC07_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC08 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid08, id_lineage08['species'], read_count08)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count08)*100\n",
    "\n",
    "unclassified_count = total_count08 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC08_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC09 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid09, id_lineage09['species'], read_count09)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count09)*100\n",
    "\n",
    "unclassified_count = total_count09 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC09_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC10 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid10, id_lineage10['species'], read_count10)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count10)*100\n",
    "\n",
    "unclassified_count = total_count10 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC10_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC11 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid11, id_lineage11['species'], read_count11)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count11)*100\n",
    "\n",
    "unclassified_count = total_count11 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC11_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n",
    "\n",
    "\n",
    "##########################\n",
    "# BC12 species: Read count \n",
    "\n",
    "species = pd.DataFrame(list(zip(taxid12, id_lineage12['species'], read_count12)), \n",
    "                            columns=['taxid', 'species', 'read_count'] )\n",
    "\n",
    "\n",
    "count_species = pd.DataFrame(species.groupby('species')['read_count'].sum())\n",
    "count_species.insert(0, 'name', '')\n",
    "\n",
    "name = []\n",
    "for i in count_species.index.values:\n",
    "    \n",
    "    name.append(list(ncbi.get_taxid_translator([i]).values())[0])\n",
    "    \n",
    "    \n",
    "count_species['name'] = name\n",
    "count_species = count_species.sort_values(by=['read_count'], ascending=False)\n",
    "\n",
    "count_species['percentage'] = (count_species['read_count'] / total_count12)*100\n",
    "\n",
    "unclassified_count = total_count12 - count_species.sum(axis = 0, skipna = True)[1]\n",
    "unclassified_per = 100 - count_species.sum(axis = 0, skipna = True)[2]\n",
    "count_species = count_species.append({\"name\": \"Unclassified\", \"read_count\": unclassified_count,\"percentage\":unclassified_per}, ignore_index=True)\n",
    "\n",
    "count_species.to_csv(result_path + 'BC12_count_species_noSelect.csv', encoding='utf-8')\n",
    "count_species\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
